ValidationMixin.js

```diff
@@ -1,83 +1,53 @@
-      state: this.state,
-      field: field
-    };
-    var nextErrors = assign({}, this.state.errors, ValidationFactory.validate(options));
-    this.setState({
-      errors: nextErrors
-    });
-    return nextErrors;
   },
 
   /**
-   * Convenience method to validate a field via an event handler. Useful for
-   * onBlur, onClick, onChange, etc...
+   * Validate single form field or entire form against the component's state.
    *
-   * @param {?String} field State key to validate
-   * @param {?Boolean} preventDefault flag to indicate that this event should be canceled.
-   * default is false.
-   * @return {function} validation event handler
+   * @param {?String} key. Field key to validate (entire form if undefined).
+   * @return {Boolean} Result of `isValid` call after validation.
    */
-  handleValidation: function(field, preventDefault) {
-    return function(event) {
-      if (preventDefault === true) {
-        event.preventDefault();
-      }
-      this.validate(field);
-    }.bind(this);
+  validate: function(key) {
+    var schema = result(this, 'validatorTypes', {});
+    var errors = Object.assign({}, this.state.errors, ValidationFactory.validate(this.state, schema, key));
+    this.setState({
+      errors: errors
+    });
+    return this.isValid(key);
   },
 
   /**
-   * Returns all validation messages for a single field, or all fields if
-   * no field is provided.
+   * Get current validation messages for a specified field or entire form.
    *
-   * @param {?String} field State key to validate
-   * @return {Array} all validation messages
+   * @param {?String} key. Field key to get messages (entire form if undefined)
+   * @return {Array}
    */
-  getValidationMessages: function(field) {
-    return ValidationFactory.getValidationMessages(this.state.errors, field);
+  getValidationMessages: function(key) {
+    return ValidationFactory.getValidationMessages(this.state.errors, key);
   },
 
   /**
-   * Determines if the current components state is valid. This method is lazy
-   * if a field is specified. This allows developers to check if errors have
-   * been reported for this field without forcing a revalidation. If no field
-   * is provided the entire form will be forcefully revalidated.
+   * Check current validity for a specified field or entire form.
    *
-   * @param {?String} field State key to validate
-   * @return {Boolean} returns validity of single field or entire form
+   * @param {?String} key. Field key to check validity (entire form if undefined).
+   * @return {Boolean}.
    */
-  isValid: function(field) {
-    if (field) {
-      //validate single field only; lazy validation
-      return ValidationFactory.isValid(this.state.errors, field);
-    } else {
-      //force full form validation if no field is provided
-      return ValidationFactory.isValid(this.validate(), field);
-    }
+  isValid: function(key) {
+    return ValidationFactory.isValid(this.state.errors, key);
```

Comment:
How is the user able to validate the entire form? If a user wants to validated all fields `onSubmit` they currently can't do this since `isValid` assumes previous validation actions occurred. For example if a user clicks Submit on an empty form, `this.state.errors` will be empty and the form will be valid...which it is not. That was the difference between Lazy and Eager validation. We need a way for the user to validate the complete form.